test/sorting.jl

```diff
@@ -299,6 +299,61 @@ Base.step(r::ConstantRange) = 0
         end
     end
 end
+@testset "insorted" begin
+    numTypes = [ Int8,  Int16,  Int32,  Int64,  Int128,
+                UInt8, UInt16, UInt32, UInt64, UInt128,
+                Float16, Float32, Float64, BigInt, BigFloat]
+
+    @test insorted(1, collect(1:10), by=(x -> x >= 5)) == true
+    @test insorted(10, collect(1:10), by=(x -> x >= 5)) == true
+
+    for R in numTypes, T in numTypes
+        @test insorted(T(0), R[1, 1, 2, 2, 3, 3]) === false
+        @test insorted(T(1), R[1, 1, 2, 2, 3, 3]) == true
+        @test insorted(T(2), R[1, 1, 2, 2, 3, 3]) == true
+        @test insorted(T(4), R[1, 1, 2, 2, 3, 3]) === false
+        @test insorted(2.5, R[1, 1, 2, 2, 3, 3]) === false
+
+        @test insorted(T(0), 1:3) === false
+        @test insorted(T(1), 1:3) == true
+        @test insorted(T(2), 1:3) == true
+        @test insorted(T(4), 1:3) === false
+
+        @test insorted(T(1), R(collect(1:10)), by=(x -> x >= 5)) == true
+        @test insorted(T(10), R(collect(1:10)), by=(x -> x >= 5)) == true
```

Comment:
```suggestion
        @test insorted(T(1), R.(collect(1:10)), by=(>=(5)))
        @test insorted(T(10), R.(collect(1:10)), by=(>=(5)))
```
1. For the `BigFloat` case, you need to apply the conversion elementwise, you can't hit the entire collection with it.
2. `>=(5)` is a short-hand for `x -> x >=5`.
3. Since `insorted` returns a `Bool`, there is no need to compare to true.
4. 2.-3. are just style comments.

Reply:
1. It just made me understand why `R[1:10;]` is preferred over collect() in other tests. Thanks!
2. 3. This is much better with these changes.